sdk/identity/identity/src/credentials/interactiveBrowserCredential.ts

```diff
@@ -116,71 +120,101 @@ export class InteractiveBrowserCredential implements TokenCredential {
     // eslint-disable-next-line
     return new Promise<AccessToken | null>(async (resolve, reject) => {
       // eslint-disable-next-line
-      let listen: http.Server | undefined;
       let socketToDestroy: Socket | undefined;
 
-      function cleanup(): void {
-        if (listen) {
-          listen.close();
-        }
-        if (socketToDestroy) {
-          socketToDestroy.destroy();
+      const requestListener = (req: http.IncomingMessage, res: http.ServerResponse) => {
+        if (!req.url) {
+          reject(
+            new Error(
+              `Interactive Browser Authentication Error "Did not receive token with a valid expiration"`
+            )
+          );
+          return;
         }
-      }
-
-      // Create Express App and Routes
-      const app = express();
-
-      app.get("/", async (req, res) => {
+        const url = new URL(req.url, this.redirectUri);
```

Comment:
I'd probably put a try/catch around everything before calling into `acquireTokenByCode` so you can reject with an Error if URL throws or something